yt/frontends/parthenon/data_structures.py

```diff
@@ -0,0 +1,334 @@
+        component_name_offset = 0
+        for dname, num_component in zip( dnames, num_components):
+            for j in range(num_component):
+                fname = self._handle["Info"].attrs["ComponentNames"][j + component_name_offset]
+                self._field_map[fname] = (dname, j)
+                k += 1
+            component_name_offset  = int(component_name_offset + num_component)
+
+
+        self.refine_by = 2
+        dimensionality = 3
+        if self.domain_dimensions[2] == 1:
+            dimensionality = 2
+        if self.domain_dimensions[1] == 1:
+            dimensionality = 1
+        self.dimensionality = dimensionality
+        self.current_time = self._handle["Info"].attrs["Time"]
+        self.cosmological_simulation = False
+        self.num_ghost_zones = 0
+        self.field_ordering = "fortran"
+        self.boundary_conditions = [1] * 6
+
+        if "periodicity" in self.specified_parameters:
+            self._periodicity = tuple(
+                self.specified_parameters["periodicity"])
+        else:
+            boundary_conditions = self._handle["Info"].attrs["BoundaryConditions"]
+
+            inner_bcs = boundary_conditions[::2]
+            #outer_bcs = boundary_conditions[1::2]
+            ##Check self consistency
+            #for inner_bc,outer_bc in zip(inner_bcs,outer_bcs):
+            #    if( (inner_bc == "periodicity" or outer_bc == "periodic") and inner_bc != outer_bc ):
+            #        raise Exception("Inconsistent periodicity in boundary conditions")
+
+            self._periodicity = tuple((bc == "periodic" for bc in inner_bcs))
+
+
+        gamma_attrs = {key:val for key, val in self._handle["Params"].attrs.items() if key.endswith("AdiabaticIndex")} 
+        if "gamma" in self.specified_parameters:
+            self.gamma = float(self.specified_parameters["gamma"])
+        elif len(gamma_attrs) >= 1 :
+            self.gamma = list(gamma_attrs.values())[0]
+        else:
+            self.gamma = 5./3.
+
+
+        He_mass_fraction_attrs = {key:val for key, val in self._handle["Params"].attrs.items() if key.endswith("He_mass_fraction")} 
+        if len(He_mass_fraction_attrs) >= 1 :
+            He_mass_fraction = list(He_mass_fraction_attrs.values())[0]
+            H_mass_fraction = 1.0 - self.He_mass_fraction
+            self.mu = 1 / (He_mass_fraction * 3. / 4. + (1 - He_mass_fraction) * 2)
+        else:
+            He_mass_fraction =  _primordial_mass_fraction["He"]
+            H_mass_fraction  =  _primordial_mass_fraction["H"]
+            self.mu = self.specified_parameters.get(
+                "mu", compute_mu(self.default_species_fields)
+            )
+
+        self.parameters["He_mass_fraction"] = He_mass_fraction
+        self.parameters["H_mass_fraction"]  = H_mass_fraction
+
+        self.current_redshift = (
+            self.omega_lambda
+        ) = (
+            self.omega_matter
+        ) = self.hubble_constant = self.cosmological_simulation = 0.0
```

Comment:
readability ++
(and the `cosmological_simulation` attribute should be an `int`)
```suggestion
        self.current_redshift = 0.0
        self.omega_lambda = 0.0
        self.omega_matter = 0.0
        self.hubble_constant = 0.0
        self.cosmological_simulation = 0
```